batch/batch/server/server.py

```diff
@@ -132,8 +173,25 @@ def _create_pod(self):
                                                                    self.id,
                                                                    self._current_task.name))
 
-    def _delete_pod(self):
-        if self._pod_name:
+    def _delete_pvc(self):
+        if self._pvc is not None:
+            try:
+                v1.delete_namespaced_persistent_volume_claim(
+                    self._pvc.metadata.name,
+                    POD_NAMESPACE,
+                    kube.client.V1DeleteOptions(),
+                    _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
+            except kube.client.rest.ApiException as err:
+                if err.status == 404:
+                    log.info(f'persistent volume claim {self._pvc.metadata.name} in '
+                             f'{self._pvc.metadata.namespace} is already deleted')
+                else:
+                    log.warning(f'persistent volume claim {self._pvc.metadata.name} in '
```

Comment:
I think that at least some subset of errors should be returned to users. 404 is fine not to return, that is effectively 200 from the perspective of the delete action. But a 500-ish error I may want to know about, if anything on that volume is sensitive.

Reply:
sure